velox/functions/sparksql/tests/ArrayMaxTest.cpp

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <limits>
+#include <optional>
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::functions::test;
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+class ArrayMaxTest : public SparkFunctionBaseTest {
+ protected:
+  void testArrayMax(const VectorPtr& input, const VectorPtr& expected) {
+    auto result = evaluate<BaseVector>("array_max(C0)", makeRowVector({input}));
+    assertEqualVectors(expected, result);
+  }
+};
+
+TEST_F(ArrayMaxTest, booleanWithNulls) {
+  auto input = makeNullableArrayVector<bool>(
+      {{true, false},
+       {true},
+       {false},
+       {},
+       {true, false, true, std::nullopt},
+       {std::nullopt, true, false, true},
+       {false, false, false},
+       {true, true, true}});
+
+  auto expected = makeNullableFlatVector<bool>(
+      {true, true, false, std::nullopt, true, true, false, true});
+
+  testArrayMax(input, expected);
+}
+
+TEST_F(ArrayMaxTest, booleanNoNulls) {
```

Comment:
No need to split with-null and no-nulls test. Let's make a test per type.

Reply:
done